tests/test_odcs_compose_generator.py

```diff
@@ -0,0 +1,135 @@
+        """path to container.yaml with content"""
+        path = tmp_path / "container.yaml"
+        with path.open("w") as file:
+            yaml.dump(container_data, file)
+        return path
+
+    @pytest.fixture()
+    def content_sets_data(self) -> dict:
+        """content_sets.yaml content"""
+        return {
+            "x86_64": [
+                "rhel-8-for-x86_64-baseos-eus-rpms__8_DOT_6",
+                "rhel-8-for-x86_64-appstream-eus-rpms__8_DOT_6",
+            ],
+            "aarch64": [
+                "rhel-8-for-aarch64-baseos-eus-rpms__8_DOT_6",
+                "rhel-8-for-aarch64-appstream-eus-rpms__8_DOT_6",
+            ],
+        }
+
+    @pytest.fixture()
+    def content_sets_yaml(self, content_sets_data: dict, tmp_path: Path) -> Path:
+        """path to content_sets.yaml with content"""
+        path = tmp_path / "content_sets.yml"
+        with path.open("w") as file:
+            yaml.dump(content_sets_data, file)
+        return path
+
+    @pytest.fixture()
+    def compose_path(self, tmp_path: Path) -> Path:
+        """Path to where the compose should be stored"""
+        return tmp_path / "repofile.repo"
+
+    @pytest.fixture()
+    def mock_config_generator(self, monkeypatch: MonkeyPatch) -> MagicMock:
+        """Monkey-patched ODCSConfigurationsGenerator"""
+        mock: MagicMock = create_autospec(
+            ODCSConfigurationsGenerator,
+            return_value=create_autospec(ODCSConfigurationsGenerator, instance=True),
+        )
+        monkeypatch.setattr(
+            odcs_compose_generator, ODCSConfigurationsGenerator.__name__, mock
+        )
+        return mock
+
+    @pytest.fixture()
+    def mock_compose_generator(self, monkeypatch: MonkeyPatch) -> MagicMock:
+        """Monkey-patched ComposeGenerator"""
+        mock: MagicMock = create_autospec(ComposeGenerator)
+        monkeypatch.setattr(odcs_compose_generator, ComposeGenerator.__name__, mock)
+        return mock
+
+    @pytest.fixture()
+    def mock_requester(self, monkeypatch: MonkeyPatch) -> MagicMock:
+        """Monkey-patched ODCSRequester"""
+        mock: MagicMock = create_autospec(ODCSRequester)
+        monkeypatch.setattr(odcs_compose_generator, ODCSRequester.__name__, mock)
+        return mock
+
+    @pytest.fixture()
+    def mock_fetcher(self, monkeypatch: MonkeyPatch) -> MagicMock:
+        """Monkey-patched ODCSFetcher"""
+        mock: MagicMock = create_autospec(ODCSFetcher)
+        monkeypatch.setattr(odcs_compose_generator, ODCSFetcher.__name__, mock)
+        return mock
+
+    def test_main(  # pylint: disable=too-many-arguments
+        self,
+        compose_path: Path,
+        container_data: dict,
+        container_yaml: Path,
+        content_sets_data: dict,
+        content_sets_yaml: Path,
+        mock_compose_generator: MagicMock,
+        mock_config_generator: MagicMock,
+        mock_requester: MagicMock,
+        mock_fetcher: MagicMock,
```

Comment:
I wonder if that heavy usage of monkeypatching may be hinting us at a future dependency injection refactor.

Reply:
We are using dependency injection. Those tests for the entrypoint script is the only place where we'd have to use monkeypatch, but I can see if I can make those as inputs to the script as well.

Reply:
I ran into some trouble passing the injected dependency to the script main function during tests, which also needs to take click command line arguments.
We did some somewhat similar things in the past, but for some reason the unit test invoked the function with just the command line arguments and not the injected dependencies.

As the current pattern of using monkeypatch only on the command line script is what we did in the past as well, I'm going to leave it as-is for now.